fcrepo-exts/fcrepo-camel-toolbox
# Upgrades audit triplestore service to Camel 3.9.0.

**Upgrades audit triplestore service to Camel 3.9.0.**
* * *

**JIRA Ticket**: https://fedora-repository.atlassian.net/browse/FCREPO-3761

* Other Relevant Links (Mailing list discussion, related pull requests, etc.)

# What does this Pull Request do?
Brings the audit triplestore service in line with other camel toolbox microservices by
1) removing osgi dependencies
2) upgrade to Camel 3.9
3) audit service can be enabled using the `audit.enabled=true` property.


# What's new?

# How should this be tested?
in your fcrepo-camel-toolbox.properties file:
```
audit.enabled=true
audit.triplestore.baseUrl=http://localhost:3030/test
```

Start fedora: 

```
docker run --rm -p8080:8080 -p61616:61616  -p8181:8181 --name=my_fcrepo6  fcrepo/fcrepo:latest
```

start fuseki:
```
 docker run --rm -p 3030:3030 --name my_fuseki atomgraph/fuseki --mem /test
```

start camel toolbox:

```
java -jar fcrepo-camel-toolbox-app-6.0.0-SNAPSHOT-driver.jar -c fcrepo-camel-toolbox.properties
```

Add a binary resource
```
curl -u fedoraAdmin:fedoraAdmin http://localhost:8080/fcrepo/rest/hello-world  -H "Content-Type: text/plain"  -X PUT --data "hello world"
```

Verify the results in the triplestore by opening the following link in your browser:
```
http://localhost:3030/test/query?query=SELECT%20%2A%20WHERE%20%7B%20%3Fs%20%3Fo%20%3Fp%20.%20%7D
```

# Interested parties
Tag (@ mention) interested parties or, if unsure, @fcrepo/committers
